### PR TITLE
docs: rewrite test_to_doc_links how-to for C++, Rust, and Python

### PR DESCRIPTION
**Pull request by a-zw**
See https://eclipse-score.github.io/docs-as-code/pr-549/how-to/test_to_doc_links.html

### CODE REVIEW

File: docs/how-to/test_to_doc_links.rst [hunk 1]
```diff
@@ -17,47 +17,76 @@
...
-	 from attribute_plugin import add_test_properties
+Every linked test must declare the following properties
+(see :need:`gd_guidl__verification_specification`):
 
-	 @add_test_properties(
-			 partially_verifies=["tool_req__docs_common_attr_title", "tool_req__docs_common_attr_description"],
-			 test_type="interface-test",
-			 derivation_technique="boundary-values",
-	 )
-	 def test_feature():
```
Comment (AlexanderLanin) [inline docs/how-to/test_to_doc_links.rst:36]: _AI-assisted review (Claude):_

**Important** — `Description` is listed as a mandatory property for link creation, but the linker does not enforce it.

In `xml_parser.py:160`, the `Description` property is explicitly skipped during parsing:
```python
if prop_name == "Description":
    continue
```

And `DataOfTestCase` in `testlink.py` has no `Description` field — so `is_valid()` never checks for it. A test that omits `Description` will still be linked successfully.

The statement on line 47 — *"If any mandatory property is missing the test will be skipped during link creation"* — is technically incorrect for `Description`. It applies to `PartiallyVerifies`/`FullyVerifies`, `TestType`, and `DerivationTechnique`, but not to `Description`.

`Description` may well be a *process* requirement (from the verification specification), but the way it's presented here implies the linker enforces it, which it doesn't.

**Suggested fix:** Either clarify that `Description` is a process requirement (not enforced by the linker), or split the list into linker-enforced vs. process-required properties.

**Review by AlexanderLanin — COMMENTED** [review]
_AI-assisted review (Claude):_

**Severity breakdown:** Critical: 0, Important: 1, Suggestions: 1

Nice rewrite — the structure is much clearer and the language-specific guidance with cross-references to requirements is a solid improvement over the old Python-only example.

**One accuracy issue with the `Description` property claim** (see inline comment) and **one minor discoverability note** about a dropped cross-reference to `dashboards_and_quality_gates`. The old doc had a "Related" section linking there — might be worth keeping as a brief "See also" since it covers end-to-end dashboard and CI threshold setup, which is a natural next step after setting up test links.

The two external need references (`gd_guidl__verification_specification`, `gd_temp__verification_specification`) appear to resolve correctly in the PR preview build, so those look fine.

File: docs/how-to/test_to_doc_links.rst (hunk 1 above)
Comment (a-zw) [reply docs/how-to/test_to_doc_links.rst:36]: I consider that a bug in our implementation but this documentation is fine.

**Review by AlexanderLanin — APPROVED** [review]
